alkos/kernel/src/kernel.cpp

```diff
@@ -14,12 +6,26 @@
 #error "AlkOS needs to be compiled with a x86_64-elf compiler"
 #endif
 
-/* Make kernel entry C compatible */
-extern "C"
-{
+#ifdef ALKOS_TEST
+#include "../test/tester.hpp"
+#endif
+
+#include "../include/init.hpp"
+
+static void KernelRun() {
+}
+
+extern "C" const char *KernelMain() {
+    KernelInit();
+
+#ifdef ALKOS_TEST
+
+    VERIFY_TEST_TYPE(ALKOS_TEST)
+    RunTest(static_cast<TestType>(ALKOS_TEST));
+    return "Hello from AlkOS test!";
```

Comment:
I think we could at least move this from main to some function e.g. RunAllTests
and just wrap the one function in ifdef for code clarity. But idk what are your plans for the future. For me something doesn't feel right about the current approach (inlining tests)

Reply:
Not applicable to this PR, we should have some clear vision how tests will be done its too foggy right now.